vsphere/config_test.go

```diff
@@ -73,7 +73,6 @@ func testAccClientPreCheck(t *testing.T) {
 func testAccClientGenerateConfig() *Config {
 	insecure, _ := strconv.ParseBool(os.Getenv("TF_VAR_VSPHERE_ALLOW_UNVERIFIED_SSL"))
 	debug, _ := strconv.ParseBool(os.Getenv("TF_VAR_VSPHERE_CLIENT_DEBUG"))
-
```

Comment:
Remove the change since it brings no value.

Reply:
Hello @vasilsatanasov ,
Thanks for looking into my MR.The reason I proceeded with this MR is that the Datastore data refer to a single datastore, whereas I need information on all datastores. This can be usefull in cases of migration, creation of new VMs and balancing in general. 

I am aware that this could overlap with Storage DRS in some parts, however this implementation is a terraform-side implementation and opens up a wide variety of angles to handle the selection of a datastore.

There could be more use cases but this is the basic one of our use.

Thanks,
Nikos

Reply:
Hi @nikfot I probably did not state my question correctly. Since you added the stats property to the `data "vsphere_datastore"` why do you need additional data (`data "datastore_stats"`) to select a datastore based on the stats ? The stats of the datastore can be obtained from  `data "vsphere_datastore"` without having additional data added to the provider source (`data "datastore_stats"`) so additional data source looks redundant (i.e. overlaping with the data provided in`data "vsphere_datastore".stats` property). 
That's why I am asking what is the business case that `data "vsphere_datastore".stats` property can not cover.

Regards, Vasil

Reply:
Hi,
The vsphere_datastore object is  providing info on a specific datastore. To use the stats object in it, you should get the specific datastore and evaluate it. However, it can be usefull for checking if a specific datastore matches some criteria.

Now the vsphere_datastore_stats struct is a purely "data" struct that can help you choose from a list, without knowing before hand the name(s) of specific datastore(s).

Let me know what you think.

Thanks,
Nikos

Reply:
Hi Nikos, could you provide a specific HCL how you select datastore based on capacity using vsphere_datastore_stats data ?

Reply:
Hello @vasilsatanasov ,

Sorry for the delay I had missed your comment. Here is a small example:
main.tf:
```
data "vsphere_datacenter" "dc" {
  name = var.vsphere_dc
}
data "vsphere_datastore_stats" "datastore_stats" {
  datacenter_id = data.vsphere_datacenter.dc.id
}
```

locals.tf:
```
locals {
  free_space_values   = { for k, v in data.vsphere_datastore_stats.datastore_stats.free_space : k => tonumber(v) }
  filtered_values     = { for k, v in local.free_space_values : k => tonumber(v) if v != null }
  numeric_values      = [for v in values(local.filtered_values) : tonumber(v)]
  max_free_space      = max(local.numeric_values...)
  max_free_space_name = [for k, v in local.filtered_values : k if v == local.max_free_space][0]
}
```

providers.tf:
```
provider "vsphere" {
  user                 = var.vcenter_name
  password             = var.vcenter_password
  vsphere_server       = var.vsphere_server_bt
  allow_unverified_ssl = var.allow_unverified_ssl
}

terraform {
  required_providers {
    vsphere = {
      source  = "localhost/upstream/vsphere"
      version = "1.0.0"
    }
  }
}
```

outputs.tf:
```
output "max_free_space_name" {
  value = local.max_free_space_name
}

output "max_free_space" {
  value = local.max_free_space
}
```

variables.tf:
```

variable "vcenter_name" {
  description = "Vsphere username"
  type        = string
}

variable "vcenter_password" {
  description = "Password for Vcenter"
  type        = string
}

variable "allow_unverified_ssl" {
  description = "Allow unverified ssl(true or false)"
  type        = bool
}


variable "vsphere_server_bt" {
  description = "Vcenter server for BT"
  type        = string
}



variable "vsphere_dc" {
  description = "Name of the vsphere datacenter you want to deploy the VM to."
  type        = string
}
```

So this is how I plan with the current change to get the best suitable datastore. In general I have more internal changes to the provider that I would like to contribute if this is accepted, and they will make more sense and provide a more complete functionality.

Let me know what you think.
Thanks,
Nikos

Reply:
The business justification looks OK for me. There is no way to select datastore by stats without knowing the datastore name. As per different contributions - this is an open source project , you are free and welcome to contribute!

Reply:
Ok thanks for the feedback. So this is ok? Will it be included in one of the future releases?

Reply:
Probably yes. Depends on the release management. I can not commit for any specific release or date. Also this PR should not be a draft. Draft PRs cannot be merged acording ot [docs](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests#draft-pull-requests).

Reply:
Ok removed this from drafts. is there anything else I can do regarding this PR or I am just waiting for final feedback?